frontend/pkg/frontend/frontend.go

```diff
@@ -284,92 +284,102 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 
-	versionedInterface, err := VersionFromContext(ctx)
+	if updating {
+		f.logger.Info(fmt.Sprintf("document already exists for %s, use PATCH instead", doc.Key))
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	versionedRequestCluster := versionedInterface.NewHCPOpenShiftCluster(nil)
+	body, err := BodyFromContext(ctx)
 	if err != nil {
 		f.logger.Error(err.Error())
 		arm.WriteInternalServerError(writer)
 		return
 	}
+	if err = json.Unmarshal(body, versionedRequestCluster); err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteCloudError(writer, arm.NewUnmarshalCloudError(err))
+		return
+	}
 
-	systemData, err := SystemDataFromContext(ctx)
+	hcpCluster := api.NewDefaultHCPOpenShiftCluster()
+	versionedRequestCluster.Normalize(hcpCluster)
+	hcpCluster.Name = request.PathValue(PathSegmentResourceName)
+
+	newCsCluster, err := f.BuildCSCluster(ctx, hcpCluster)
 	if err != nil {
 		f.logger.Error(err.Error())
 		arm.WriteInternalServerError(writer)
 		return
 	}
 
-	f.logger.Info(fmt.Sprintf("%s: ArmResourceCreateOrUpdate", versionedInterface))
+	req, err := f.PostCSCluster(newCsCluster)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+	doc.ClusterID = req.Body().ID()
+	err = f.dbClient.SetClusterDoc(ctx, doc)
+	if err != nil {
+		f.logger.Error(fmt.Sprintf("failed to create document for resource %s: %v", doc.ClusterID, err))
+	}
 
-	// URL path is already lowercased by middleware.
-	resourceID := request.URL.Path
-	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	resp, err := json.Marshal(versionedRequestCluster)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+	writer.WriteHeader(http.StatusCreated)
+	_, err = writer.Write(resp)
+	if err != nil {
+		f.logger.Error(err.Error())
+	}
+}
 
-	var doc *database.HCPOpenShiftClusterDocument
-	var updating bool = true
-	doc, err = f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+func (f *Frontend) ArmResourceUpdate(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+	versionedInterface, doc, updating, err := f.getVersionedInterfaceAndDoc(ctx, writer, request)
 	if err != nil {
-		if errors.Is(err, database.ErrNotFound) {
-			updating = false
-			f.logger.Info(fmt.Sprintf("existing document not found for cluster - creating one for %s", resourceID))
-			doc = &database.HCPOpenShiftClusterDocument{
-				ID:           uuid.New().String(),
-				Key:          resourceID,
-				PartitionKey: subscriptionID,
-				SystemData:   systemData,
-			}
-		} else {
-			f.logger.Error(fmt.Sprintf("failed to fetch document for %s: %v", resourceID, err))
-			arm.WriteInternalServerError(writer)
-			return
-		}
+		return
+	}
+
+	if !updating {
+		originalPath, _ := OriginalPathFromContext(ctx)
+		f.logger.Error("Resource not found")
+		arm.WriteError(writer, http.StatusNotFound, arm.CloudErrorCodeNotFound, originalPath, "Resource not found")
+		return
```

Comment:
Similar to above, but in this scenario if the doc is not found its likely the cluster doesnt exist and patching just fails so the error check just above this may handle that for us